bumble/host.py

```diff
@@ -514,7 +516,7 @@ def send_hci_packet(self, packet: hci.HCI_Packet) -> None:
         if self.hci_sink:
             self.hci_sink.on_packet(bytes(packet))
 
-    async def send_command(self, command, check_result=False):
+    async def send_command(self, command, check_result=False, timeout=None):
```

Comment:
`send_command` is async, so we don't need to add a `timeout` parameter to it in order to wait at most some amount of time: we can leave that up to the caller, which can invoke `await asyncio.wait_for(host.send_command(...), timeout)`.

Reply:
Sorry, I'm a bit new to AsyncIO. If the calling thread times out on the co-routine, while the co-routine terminate and yield the lock so that another `send_command()` can run?

Reply:
with `wait_for`, if the timeout expires before the coroutine returns, the coroutine is canceled, which means that here the `await self.pending_response` with raise a TimeoutError, which will exit out of the `async with self.command_semaphore`, releasing the semaphore, so if another task was waiting on it, it can now proceed.

The only small different between putting the `wait_for` on the outside, rather than just around `asyncio.wait_for(self.pending_response,...)` is that the timeout encompasses not only waiting for the response, but also waiting for the semaphore before that.
So, maybe there's a case to be made for specifying an inner timeout, like you have it, which only applies to waiting for the response. I can see how this could make sense, even if I don't have an actual use case in mind. So maybe we leave things the way you have them here, but rename the `timeout` parameter to `response_timeout`, to make it clear that it's not a timeout on the whole method, but just on the waiting for a response inside that method.

Reply:
I can do that!
- [ ] Rename `timeout -> response_timeout`